pkg/server/localproxy.go

```diff
@@ -133,27 +136,47 @@ func WithLocalProxy(
 		}
 
 		// lookup in our local, potentially partial index
-		r := indexState.Lookup(path)
-		if r.Found && r.Shard != shardName {
+		r, found := indexState.Lookup(path)
+		if found && r.Shard != shardName && r.URL == "" {
 			logger.WithValues("cluster", cluster.Name, "requestedShard", r.Shard, "actualShard", shardName).Info("cluster is not on this shard, but on another")
 
 			w.Header().Set("Retry-After", fmt.Sprintf("%d", 1))
 			http.Error(w, "Not found on this shard", http.StatusTooManyRequests)
 			return
 		}
 
+		if r.URL != "" {
+			url, err := url.Parse(r.URL)
+			if err != nil {
+				logger.WithValues("cluster", cluster.Name, "url", r.URL).Error(err, "invalid url")
+				http.Error(w, "invalid url", http.StatusInternalServerError)
+				return
+			}
+			logger.WithValues("from", path, "to", r.URL).V(4).Info("mounting cluster")
+			proxy := httputil.NewSingleHostReverseProxy(url)
+			//TODO(MVP): remove this once we have a real cert wired in dev mode
+			proxy.Transport = &http.Transport{
+				TLSClientConfig: &tls.Config{
+					InsecureSkipVerify: true,
+				},
+			}
+
+			proxy.ServeHTTP(w, req)
```

Comment:
Does the front-proxy need the same?

Reply:
I think front proxy handles this by default as It return the url in `Result`. Will double check.

Reply:
right, might be. It's proxying every request.

Reply:
yes, and it already can hook certificates for custom workspace paths.